BCStudentSoftwareDevTeam/lsf
# Added endpoint for CELTS to hit to get CELTS labor info

*There is no LSF issue attached to this PR.

## Issue: 
CELTS would like to have their labor students position title and time worked at CELTS displayed on their `User Profile` and in their `Service Transcript`

## Did: 
- Used `flask_restful` to create two simple REST GET endpoints to retrieve either an entire orgs labor forms or a specific students labor forms for a specific org.  

## To Test:
- Checkout `createCeltsEndpoint` branch and run `source setup.sh`. *There are new dependencies so you will need to rerun `setup.sh` if you have already had LSF up. 
- Reset your database with test data and run the tests. `database/reset_database.sh test`, `tests/run_tests.sh`
- Reset you database from backup. `database/reset_database.sh from-backup`
- Run the application and go to the endpoints and verify they work. 

## Changes: 
- Added the Flask extension `flask_restful`.
- Created a new controller file, `api_routes`, which contains `routes.py`, which is where the two api routes are initialized, and `apiClasses.py`, which contains two new classes, 'LaborFormsForDepartmentApi` and `LaborFormsForStudentApi`.
- The two new classes have similar functions, they both contain two simple get requests and some simple validation/authorization to make sure not just anyone can hit the endpoints. 
- Created a new logic file, `apiEndpoint.py`, which contains the `getLaborInformation` function. The function has peewee queries and logic to retrieve an org or students labor form and returns a json response object. 
- Created a new test file, `test_apiEndpoint.py`, which tests the new `getLaborInformation` function.
